### DISCUSSION TITLE
How to configure Redux

### DISCUSSION THREAD

**Discussion by Rahul-0108** [iTwin/itwinjs-core#2548] [Q&A] [answered]
I want to configure Redux in iTwinViewer , I followed these documentations :  https://www.itwinjs.org/learning/ui/hostappui/#statemanager-and-reducerregistry
https://www.itwinjs.org/learning/ui/framework/state/#example-of-defining-dynamic-reducer-needed-by-a-plugin
It mentions not to create the store but to dynamically register reducers at runtime (which I did in the event handler oniModelConnected calling the SampleExtensionStateManager.initialize() ) .
 After updating the state variable SampleExtensionStateManager.isExtensionUiVisible , the another component is not being rerendering with the new Value . I added mapStateToprops also in the another component , but still not getting the state value

**Comment by bsteinbk** [comment, marked as answer]
@aruniverse @kckst8 - Do you guys know if the itwin-view app initializes the StateManger per the instructions [here](https://www.itwinjs.org/learning/ui/framework/state/#example-of-defining-dynamic-reducer-needed-by-a-plugin) so items providers can add their own redux entries?

**Reply by aruniverse** [reply]
I do believe we properly intialize the StateManager. I have an example where I was able to register my own reducer using the ReducerRegistryInstance Singleton. This is with the old Bentley scoped viewer but should still work with the new one. 

https://github.com/aruniverse/mytwin-viewer-samples/blob/MarkerTools/src/ui/MarkerTool/UiProvider.tsx

**Comment by Rahul-0108** [comment]
Hi , it worked following the above Sample , Thanks !!
One thing I want to know can we not configure our own store anyway ??

**Reply by bsteinbk** [reply]
There can only be one Redux store per iModelApp.  With code like `ReducerRegistryInstance.registerReducer("markers", markerSlice.reducer);` you just add your state/reducer to the application's  store.  See more info [here](https://github.com/imodeljs/imodeljs/blob/5575e60ade8b32fd268abe585b05d1e302f8451f/docs/learning/ui/framework/State.md#L35)
